cnd/src/network/comit.rs

```diff
@@ -540,90 +512,111 @@ impl NetworkBehaviourEventProcess<oneshot_behaviour::OutEvent<finalize::Message>
 
         if state.sent_finalized && state.received_finalized {
             tracing::info!("Swap {} is finalized.", swap_id);
-            let (local_swap_id, create_swap_params) = self
+            let (local_swap_id, _local_data) = self
                 .swaps
                 .finalize_swap(&swap_id)
-                .expect("Swap should be known");
+                .expect("swap should be known");
 
-            let secret_hash = self
-                .secret_hashes
+            let remote_data = self
+                .remote_data
                 .get(&swap_id)
-                .copied()
-                .expect("must exist");
-
-            let ethereum_identity = self.ethereum_identities.get(&swap_id).copied().unwrap();
-            let lightning_identity = self.lightning_identities.get(&swap_id).copied().unwrap();
+                .cloned()
+                .expect("valid remote data");
 
-            self.events.push_back(BehaviourOutEvent::SwapFinalized {
+            self.events.push_back(BehaviourOutEvent {
                 local_swap_id,
-                swap_params: create_swap_params,
-                secret_hash,
-                ethereum_identity,
-                lightning_identity,
+                remote_data,
             });
         }
     }
 }
 
+/// All possible data to be sent to the remote node for any protocol
+/// combination.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct LocalData {
+    // Currently we only support herc20/halight swap, as we add protocol
+    // combinations the identities will be added here.
+    pub secret_hash: Option<SecretHash>,
+    pub shared_swap_id: Option<SharedSwapId>,
+    pub ethereum_identity: Option<identity::Ethereum>,
+    pub lightning_identity: Option<identity::Lightning>,
+}
+
+impl LocalData {
+    pub fn for_alice(secret_hash: SecretHash, identities: Identities) -> Self {
```

Comment:
I am not convinced this is useful as I soon as you add a new swap (bitcoin onchain) then you have to create many of those or just trash it.
I'd recommend to remove these two functions and directly construct the `LocalData` struct where needed (thanks to the `pub` keywords on the fields).

Reply:
I disagree, I added the next trading pair already (#2699) and these methods worked nicely. The make the code read nicely in my opinion.

Reply:
Can remove it if you feel super strongly though.

Reply:
I can't find your usage of `for_alice` in the new PR. Can you point me to a line or commit please?

Reply:
```
cnd/src/network.rs:303: let data = LocalData::for_alice(secret_hash, identities);
```